query-node/mappings/src/workingGroups.ts

```diff
@@ -473,6 +473,22 @@ export async function workingGroups_AppliedOnOpening({ store, event }: EventCont
   await store.save<AppliedOnOpeningEvent>(appliedOnOpeningEvent)
 }
 
+async function removeIsLeadFromGroup(store: DatabaseManager, groupId: string) {
+  const groupWorkers = await store.getMany(Worker, {
+    where: {
+      group: { id: groupId }
+    },
+    relations: ['group']
+  })
+
+  await Promise.all(
+    groupWorkers.map((worker) => {
+      worker.isLead = false
+      return store.save<Worker>(worker)
```

Comment:
Such a loop will cause many SQL statements. For large working groups this may become a performance issue. 

I would suggest to look for the possibility to do only one SQL statement: 

`update WG_WORKERS set isLead = 0 where group_id=?1`

Reply:
I didn't find out the SQL running way, so added filter option; isLead: true, so it will return one record and only execute one SQL statement.